src/main/java/com/okta/tools/helpers/SessionHelper.java

```diff
@@ -93,15 +94,15 @@ public void updateCurrentSession(Instant expiryInstant, String profileName) thro
         );
     }
 
-    public void addOrUpdateProfile(Instant start) throws IOException {
+    public void addOrUpdateProfile(Instant sessionExpiry) throws IOException {
         FileHelper.usingPath(FileHelper.getOktaDirectory().resolve("profiles"), reader -> {
             MultipleProfile multipleProfile = new MultipleProfile(reader);
-            multipleProfile.addOrUpdateProfile(environment.oktaProfile, environment.awsRoleToAssume, start);
+            multipleProfile.addOrUpdateProfile(environment.oktaProfile, environment.awsRoleToAssume, sessionExpiry);
             return multipleProfile;
         }, MultipleProfile::save);
     }
 
     public boolean sessionIsActive(Instant startInstant, Session session) {
-        return startInstant.isBefore(session.expiry);
```

Comment:
:bug: named profile sessions would result in sts:AssumeRoleWithSAMLResponse every time they were used since this logic didn't account for them.